iceoryx_posh/test/moduletests/test_capro_service.cpp

```diff
@@ -28,166 +30,653 @@ using namespace iox::capro;
+    EXPECT_THAT(std::to_string(serviceDescription1.getInstanceID()), serviceDescription1.getInstanceIDString());
 }
 
-TEST_F(ServiceDescription_test, CtorIDStrings)
+TEST_F(ServiceDescription_test, ServiceDescriptionDefaultCtorInitializesTheIDsAndStringsToZero)
 {
-    IdString_t testService("Service");
-    IdString_t testInstance("Instance");
-    IdString_t testEvent("Event");
-    ServiceDescription::ClassHash testHash = {1, 2, 3, 4};
-    ServiceDescription csd = ServiceDescription(testService, testInstance, testEvent, testHash);
+    ServiceDescription serviceDescription1 = ServiceDescription();
 
-    EXPECT_THAT(csd.getServiceIDString(), StrEq("Service"));
-    EXPECT_THAT(csd.getInstanceIDString(), StrEq("Instance"));
-    EXPECT_THAT(csd.getEventIDString(), StrEq("Event"));
-    EXPECT_EQ(uint32_t(1), csd.getClassHash()[0]);
-    EXPECT_EQ(uint32_t(2), csd.getClassHash()[1]);
-    EXPECT_EQ(uint32_t(3), csd.getClassHash()[2]);
-    EXPECT_EQ(uint32_t(4), csd.getClassHash()[3]);
+    EXPECT_EQ(serviceDescription1.getServiceID(), 0U);
+    EXPECT_EQ(serviceDescription1.getEventID(), 0U);
+    EXPECT_EQ(serviceDescription1.getInstanceID(), 0U);
+    EXPECT_THAT(serviceDescription1.getServiceIDString(), StrEq("0"));
+    EXPECT_THAT(serviceDescription1.getEventIDString(), StrEq("0"));
+    EXPECT_THAT(serviceDescription1.getInstanceIDString(), StrEq("0"));
 }
 
-TEST_F(ServiceDescription_test, operatorEq)
+TEST_F(ServiceDescription_test, ServiceDescriptionDefaultCtorInitializesTheScopeToWorldWide)
 {
-    EXPECT_TRUE(csd1 == csd1Eq);
-    EXPECT_TRUE(csd2 == csd2Eq);
-    EXPECT_TRUE(csd3 == csd3Eq);
-    EXPECT_TRUE(csd4 == csd4Eq);
+    ServiceDescription serviceDescription1 = ServiceDescription();
 
-    EXPECT_FALSE(csd1 == csd1Ne);
-    EXPECT_FALSE(csd2 == csd2Ne);
-    EXPECT_FALSE(csd3 == csd3Ne);
-    EXPECT_FALSE(csd4 == csd4Ne);
+    EXPECT_THAT(serviceDescription1.getScope(), Eq(Scope::WORLDWIDE));
 }
 
-TEST_F(ServiceDescription_test, operatorNe)
+TEST_F(ServiceDescription_test, ServiceDescriptionDefaultCtorInitializesTheInterfaceToInternal)
 {
-    EXPECT_FALSE(csd1 != csd1Eq);
-    EXPECT_FALSE(csd2 != csd2Eq);
-    EXPECT_FALSE(csd3 != csd3Eq);
-    EXPECT_FALSE(csd4 != csd4Eq);
+    ServiceDescription serviceDescription1 = ServiceDescription();
 
-    EXPECT_TRUE(csd1 != csd1Ne);
-    EXPECT_TRUE(csd2 != csd2Ne);
-    EXPECT_TRUE(csd3 != csd3Ne);
-    EXPECT_TRUE(csd4 != csd4Ne);
+    EXPECT_THAT(serviceDescription1.getSourceInterface(), Eq(Interfaces::INTERNAL));
 }
 
-TEST_F(ServiceDescription_test, operatorAssign)
+TEST_F(ServiceDescription_test, ServiceDescriptionStringCtorCreatesServiceDescriptionWithValuesPassedToTheCtor)
 {
-    ServiceDescription csdNew = csd1;
-    EXPECT_TRUE(csdNew == csd1);
+    testService = "1";
+    testInstance = "2";
+    testEvent = "3";
+    ServiceDescription::ClassHash testHash = {1U, 2U, 3U, 4U};
```

Comment:
You always use the number 1,2,3,4. This is alright but in rare cases, when we have for instance have a memory glitch and are reusing the old memory - caused by a bug - we would not discover this when we always use the same inputs. Therefore I would suggest to vary the input in every test a bit.
Add some additional integers to every number in the hash and you will be safe.